### DISCUSSION TITLE
How to get Specific Genome Annotations?

### DISCUSSION THREAD

**Discussion by chatterjee89** [stuart-lab/signac#1229] [Q&A] [answered]
Hi, following the tutorial, I'm running into an issue with genome annotation, where information regarding the nearest or overlapping gene, exon, intron, promoter, 5′ untranslated region (UTR), 3′ UTR, intergenic and other genomic features are not available. I'm trying to annotate the peaks specifically by these types. I understand that the `annotation=` parameter within the `ClosestFeature` function might be key here, but then my question becomes where do I find the annotation file that has information on these genomic feature types? Sorry if this sounds simplistic or too off-topic, but I'm just a biologist playing around with my own dataset. I'm not a trained bioinformatician but I am getting by significantly, thanks to the very user-friendly packages (and their tutorials) like Seurat and Signac!

Regards,
Deep

**Comment by timoast** [comment, marked as answer]
Genome annotation packages for most organisms are available on Bioconductor, you can check by searching for an EnsDb package for your species on Bioconductor (eg, `EnsDb.Hsapiens.v86` for human), or query the AnnotationHub database (see [here](https://www.bioconductor.org/packages/devel/bioc/vignettes/ensembldb/inst/doc/ensembldb.html#10_Getting_or_building_EnsDb_databasespackages)). Once you have an EnsDb package for your genome, you can use the function `GetGRangesFromEnsDb` in Signac to extract a set of genomic ranges from the database for downstream use. See the Signac [vignette](https://stuartlab.org/signac/articles/pbmc_vignette.html) for an example.

See [here](https://stuartlab.org/signac/articles/faq.html#an-annotation-or-genome-sequence-for-my-organism-is-not-available-on-bioconductor-what-do-i-do) for tips for if your species does not have an annotation available on Bioconductor

**Reply by chatterjee89** [reply]
Thank you for the suggestion! I will look into it - for now, I found HOMER to be a good annotation software (http://homer.ucsd.edu/homer/ngs/annotation.html).

What I did is that I extracted the `grnages(object)` as `la`
and then converted that to a bed file:

```
df <- data.frame(seqnames=seqnames(la),
  starts=start(la)-1,
  ends=end(la),
  names=c(rep(".", length(la))),
  scores=c(rep(".", length(la))),
  strands=strand(la))

write.table(df, file="foo.bed", quote=F, sep="\t", row.names=F, col.names=F)
```

Then I ran  `annotatePeaks.pl foo.bed mm10 -annStats annStats.txt > output.txt` to get the desired annotations and stats on my Signac processed data. Not very elegant, but actually very easy. Thank you for providing me with those links, I'm going to explore more of how to go about annotations from AnnotationHub.

Regards,
Deep
